drivers/sensors/ms5611.c

```diff
@@ -0,0 +1,704 @@
+ * Name: ms5611_activate
+ ****************************************************************************/
+
+static int ms5611_activate(FAR struct sensor_lowerhalf_s *lower,
+                           bool enable)
+{
+  bool start_thread = false;
+  struct ms5611_dev_s *priv = (FAR struct ms5611_dev_s *)lower;
+
+  if (enable)
+    {
+      if (!priv->enabled)
+        {
+          start_thread = true;
+        }
+    }
+
+  priv->enabled = enable;
+
+  if (start_thread == true)
+    {
+      /* Wake up the thread */
+
+      nxsem_post(&priv->run);
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: ms5611_fetch
+ ****************************************************************************/
+
+/* N.B. When fetch is enabled the sensortest doesn't respect the
+ * interval (-i) parameter, so let keep it comment until further
+ * discussion about the "issue".
+ */
+
+#if 0
+static int ms5611_fetch(FAR struct sensor_lowerhalf_s *lower,
+                        FAR char *buffer, size_t buflen)
+{
+  FAR struct ms5611_dev_s *priv = container_of(lower,
+                                               FAR struct ms5611_dev_s,
+                                               sensor_lower);
+  struct sensor_event_baro baro_data;
+
+  if (buflen != sizeof(baro_data))
+    {
+      return -EINVAL;
+    }
+
+  baro_measure_read(priv, &baro_data);
+
+  memcpy(buffer, &baro_data, sizeof(baro_data));
+
+  return buflen;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: ms5611_register
+ *
+ * Description:
+ *   Register the MS5611 character device
+ *
+ * Input Parameters:
+ *   i2c     - An instance of the I2C interface to use to communicate with
+ *             MS5611
+ *   devno   - Instance number for driver
+ *   addr    - The I2C address of the MS5611.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int ms5611_register(FAR struct i2c_master_s *i2c, int devno, uint8_t addr)
+{
+  FAR struct ms5611_dev_s *priv;
+  FAR char *argv[2];
+  char arg1[32];
+
+  int ret;
+
+  /* Initialize the MS5611 device structure */
+
+  priv = (FAR struct ms5611_dev_s *)kmm_zalloc(sizeof(struct ms5611_dev_s));
+  if (priv == NULL)
+    {
+      snerr("Failed to allocate instance\n");
+      return -ENOMEM;
+    }
+
+  priv->i2c      = i2c;
+  priv->addr     = addr;
+  priv->freq     = CONFIG_MS5611_I2C_FREQUENCY;
+  priv->interval = 1000000; /* Default interval 1s */
+
+  nxsem_init(&priv->run, 0, 0);
+  nxsem_init(&priv->exclsem, 0, 1);
+  nxsem_set_protocol(&priv->run, SEM_PRIO_NONE);
+  nxsem_set_protocol(&priv->exclsem, SEM_PRIO_NONE);
```

Comment:
Do we really need to disable priority inheritance on `exclsem`?

Reply:
I don't know for sure, I used as base other sensor driver example. Let see if it works fine without using SEM_PRIO_NONE